iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/checks/KubernetesVerifier.java

```diff
@@ -90,32 +91,66 @@ public static void verify(String templateFileName, IacCheck check) {
     String content = retrieveContent(inputFile);
     if (KubernetesParser.hasHelmContent(content)) {
       InputFileContext inputFileContext = prepareHelmContext(templateFileName);
-      verifyHelmFile(check, inputFileContext, content);
+      InternalIssueVerifier verifier = verifyHelm(check, inputFileContext, content);
+      verifier.assertOneOrMoreIssues();
     } else {
       verify(PARSER, BASE_DIR.resolve(templateFileName), check);
     }
   }
 
+  public static void verify(String templateFileName, IacCheck check, Issue... expectedIssues) {
+    InputFile inputFile = inputFile(templateFileName, BASE_DIR);
+    String content = retrieveContent(inputFile);
+    if (KubernetesParser.hasHelmContent(content)) {
+      InputFileContext inputFileContext = prepareHelmContext(templateFileName);
+      verifyHelm(check, inputFileContext, content, expectedIssues);
+    } else {
+      verify(PARSER, BASE_DIR.resolve(templateFileName), check, expectedIssues);
+    }
+  }
+
+  /**
+   * Only usable for pure Kubernetes files
+   */
   public static void verifyContent(String content, IacCheck check) {
     verify(PARSER, content, check);
   }
 
-  public static void verifyNoIssue(String fileName, IacCheck check) {
-    verifyNoIssue(PARSER, BASE_DIR.resolve(fileName), check);
+  public static void verifyNoIssue(String templateFileName, IacCheck check) {
+    InputFile inputFile = inputFile(templateFileName, BASE_DIR);
+    String content = retrieveContent(inputFile);
+    if (KubernetesParser.hasHelmContent(content)) {
+      InputFileContext inputFileContext = prepareHelmContext(templateFileName);
+      InternalIssueVerifier verifier = verifyHelm(check, inputFileContext, content);
+      verifier.assertNoIssues();
+    } else {
+      verifyNoIssue(PARSER, BASE_DIR.resolve(templateFileName), check);
+    }
   }
 
-  public static void verifyHelmFile(IacCheck check, InputFileContext inputFileContext, String content) {
+  protected static InternalIssueVerifier verifyHelm(IacCheck check, InputFileContext inputFileContext, String content) {
```

Comment:
I suggest to rename this method, because now it has the same method as `verifyHelm` that actually compares issues. This one can be `prepareHelmVerifier` or something like that